### ISSUE TITLE
Scanning bug 

### ISSUE THREAD

**Issue by adulau** [D4-project/passive-ssh#2]
```
Traceback (most recent call last):
  File "ssh_scan.py", line 259, in <module>
    res = ssh_scanner(str(v), ssh_port, use_proxy=use_proxy, proxy_ip=proxy_ip, proxy_port=proxy_port)
  File "ssh_scan.py", line 209, in ssh_scanner
    res = ssh_fingerprinter(target, ssh_port, use_proxy=use_proxy, proxy_ip=proxy_ip, proxy_port=proxy_port)
  File "ssh_scan.py", line 175, in ssh_fingerprinter
    ssh_fingerprint, host_pkey, host_ref = get_ssh_fingerprint(target, port, socket_timeout, use_proxy=use_proxy, proxy_ip=proxy_ip, proxy_port=proxy_port)
  File "ssh_scan.py", line 144, in get_ssh_fingerprint
    key = ssh_transport.get_remote_server_key()
  File "/home/adulau/analyzer-d4-passivessh/PSSHENV/lib/python3.8/site-packages/paramiko/transport.py", line 836, in get_remote_server_key
    raise SSHException("No existing session")
paramiko.ssh_exception.SSHException: No existing session
```

**Comment by adulau** [comment]
- What's the exact meaning in paramiko when the session doesn't exist
- Maybe collecting results such as "No route to host" or "Connection refused" can be useful

**Comment by adulau** [comment]
Additional issue:

```
{"banner": "SSH-2.0--", "key_exchange": {"key": ["ecdsa-sha2-nistp521", ""], "encrypt": ["aes256-ctr", "aes128-ctr"], "mac": ["hmac-sha2-256"], "compress": ["none", "zlib"], "lang": []}, "ip": "60.16.102.177", "port": 22, "date": "20201212", "epoch": 1607785906, "host_keys": [{"fingerprint": "ee:e6:70:e0:80:5c:9a:e4:b8:26:d1:b4:a7:11:c9:a0", "name": "ecdsa-sha2-nistp521", "base64": "ecdsa-sha2-nistp521 AAAAE2VjZHNhLXNoYTItbmlzdHA1MjEAAAAIbmlzdHA1MjEAAACFBAFv0Zbr23guZZAw/FMfbj8FvNZDQf7MJdjlVzPy0+vvUTmMLrqE04O0837qXbR7Uj29kco1+0mgRZeSuTUZZ8w56QHCuJyQrT5f9C9BiNx9O3PK9VlZMMb97BQ9gOUUBhRSqNd3idNbv6+M7IDbVl+j4pjid9o/oHyjW6dOhdNjTLAluw=="}, [{"banner": "SSH-2.0--", "key_exchange": {"key": ["ecdsa-sha2-nistp521", ""], "encrypt": ["aes256-ctr", "aes128-ctr"], "mac": ["hmac-sha2-256"], "compress": ["none", "zlib"], "lang": []}}, {"fingerprint": "ee:e6:70:e0:80:5c:9a:e4:b8:26:d1:b4:a7:11:c9:a0", "name": "ecdsa-sha2-nistp521", "base64": "ecdsa-sha2-nistp521 AAAAE2VjZHNhLXNoYTItbmlzdHA1MjEAAAAIbmlzdHA1MjEAAACFBAFv0Zbr23guZZAw/FMfbj8FvNZDQf7MJdjlVzPy0+vvUTmMLrqE04O0837qXbR7Uj29kco1+0mgRZeSuTUZZ8w56QHCuJyQrT5f9C9BiNx9O3PK9VlZMMb97BQ9gOUUBhRSqNd3idNbv6+M7IDbVl+j4pjid9o/oHyjW6dOhdNjTLAluw=="}, "60.16.102.177"]], "hassh": "e766215322938e4313d35b60a65063e3"}
Traceback (most recent call last):
  File "./ssh_scan.py", line 303, in <module>
    passive_ingester.save_ssh_scan(res_scan)
  File "/home/adulau/analyzer-d4-passivessh/bin/passive_ingester.py", line 59, in save_ssh_scan
    redis_ssh.sadd('all:key:type', pkey['name'])
TypeError: tuple indices must be integers or slices, not str

```

**Comment by adulau** [comment]
Traceback (most recent call last):
  File "./bin/ssh_scan.py", line 303, in <module>
    passive_ingester.save_ssh_scan(res_scan)
  File "/home/adulau/analyzer-d4-passivessh/bin/passive_ingester.py", line 59, in save_ssh_scan
    redis_ssh.sadd('all:key:type', pkey['name'])
TypeError: tuple indices must be integers or slices, not str

**Comment by adulau** [comment]
Fixed in 63c7b74010bbdb034c9388d47eb35dcf62852862 for the incorrect which could occur on large scan.
